manifests/enterprise/service/nix.pp

```diff
@@ -28,7 +34,8 @@
       command     => "${splunk::enterprise::enterprise_homedir}/bin/splunk enable boot-start ${user_args} ${splunk::params::boot_start_args} --accept-license --answer-yes --no-prompt",
       refreshonly => true,
       before      => Service[$splunk::enterprise::service_name],
-      require     => Exec['stop_splunk'],
+      require     => [Exec['stop_splunk'], Exec['disable_splunk_boot_start']],
+      subscribe   => Package[$splunk::enterprise::package_name],
```

Comment:
I think the `subscribe` isn't required because the new exec already subscribes to the same package resource?

Reply:
Actually it is needed, I just tested this using this snippet:

```puppet
file { '/tmp/foo':
  content => "foo",
}

exec { 'bar':
  command => "/usr/bin/echo bar",
  refreshonly => true,
  subscribe => File['/tmp/foo'],
}

exec { 'baz':
  command => "/usr/bin/echo baz",
  refreshonly => true,
  require => Exec['bar']
}
```

In this scenario, `baz` was never executed.

Reply:
ah right. Because `require` only enforces the ordering, but it doesn't do any refresh. 

```puppet
exec { 'enable_splunk':
    command     => "${splunk::enterprise::enterprise_homedir}/bin/splunk enable boot-start ${user_args} ${splunk::params::boot_start_args} --accept-license --answer-yes --no-prompt",
    refreshonly => true,
    before      => Service[$splunk::enterprise::service_name],
    subscribe   => Exec['disable_splunk_boot_start']
}
```

This is probably enough?

* `Exec['stop_splunk']` notifies  `Exec['enable_splunk']`, so `Exec['enable_splunk']` doesn't need to require `Exec['stop_splunk']`
* `Exec['disable_splunk_boot_start']` is already subscribed to `Package[$splunk::enterprise::package_name]`, so `Exec['enable_splunk']`  doesn't need to subscribe to `Package[$splunk::enterprise::package_name]`
* `Exec['enable_splunk']` needs to subscribe to `Exec['disable_splunk_boot_start']` for proper ordering

Reply:
I guess it kind of does the same, before it was done by subscribing to the Package, and then setting the order by adding it to the `require` parameter. 
But I changed it to the way you proposed it 😄